File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int min = 0 ;
        
        for(int i = 0 ; i < colors.length() ; i++) {
            int j = i + 1 , sum = neededTime[i] , currMax = neededTime[i] ;
            while(j < colors.length() && colors.charAt(j) == colors.charAt(i)) {
                sum += neededTime[j] ;
                currMax = Math.max(neededTime[j++] , currMax) ;
            }
            i = j - 1 ;
            min += i == j ? 0 : (sum - currMax) ;
        }

        return min ;
    }
}

File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        res , curr , mx = 0 , 0 , 0 

        for i in range(len(colors)) : 
            if i > 0 and colors[i] != colors[i-1] : 
                res += curr - mx 
                mx = curr = neededTime[i]
            else : 
                mx = max(mx, neededTime[i])
                curr += neededTime[i] 

        return res + ((curr - mx) if i > 0 and colors[i] == colors[i-1] else 0)
